field/string_test.go

```diff
@@ -37,22 +37,22 @@ func TestStringField(t *testing.T) {
 	require.Equal(t, "olleh", str.Value())
 
 	str = NewString(spec)
-	str.SetData(NewStringValue("hello"))
+	str.Marshal(NewStringValue("hello"))
 	packed, err = str.Pack()
 	require.NoError(t, err)
 	require.Equal(t, "     hello", string(packed))
 
 	str = NewString(spec)
 	data := NewStringValue("")
-	str.SetData(data)
+	str.Marshal(data)
 	length, err = str.Unpack([]byte("     olleh"))
 	require.NoError(t, err)
 	require.Equal(t, 10, length)
 	require.Equal(t, "olleh", data.Value())
```

Comment:
This test will also fail as if we remove `data` it will no longer be used in `Unpack` (internally `SetBytes`). Just test `Marshal` and `Unmarshal`.